addons/mail/models/mail_render_mixin.py

```diff
@@ -6,14 +6,15 @@
 import logging
 import re
 
+import dateutil.relativedelta as relativedelta
 from lxml import html
 from markupsafe import Markup
 from werkzeug import urls
 
 from odoo import _, api, fields, models, tools
 from odoo.exceptions import UserError, AccessError
-from odoo.tools import is_html_empty, safe_eval
-from odoo.tools.jinja import jinja_safe_template_env, jinja_template_env, template_env_globals
```

Comment:
Actually, you should keep the tools.jinja file as it was moved recently for some reason, and therefore you can keep template_env_globals where it is to lessen diff.